module/documents/actor.mjs

```diff
@@ -118,6 +118,86 @@ export default class ActorEd extends Actor {
+   * @param {object} options      Any additional options for the {@link EdRoll}.
+   */
+  async rollAbility( ability, options = {} ) {
+    const attributeStep = this.system.attributes[ability.system.attribute].step;
+    const abilityFinalStep = attributeStep + ability.system.level;
+    const difficulty = await this.getTarget( ability );
+    const edRollOptions = new EdRollOptions( {
+      rollType: "action",
+      step: { base: abilityFinalStep },
+      strain: { base: ability.system.strain},
+      target: { base: difficulty },
+      karma: { pointsUsed: this.system.karma.useAlways ? 1 : 0, available: this.system.karma.value, step: this.system.karma.step },
+      devotion: { pointsUsed: ability.system.devotionRequired ? 1: 0, pointsRequired: ability.system.devotionRequired, available: this.system.devotion.value, step: this.system.devotion.step },
+      chatFlavor: ability.name + " Test",
+    } );
+    const roll = await RollPrompt.waitPrompt( edRollOptions, options );
+    this.#processRoll( roll );
+  }
+
+  async getTarget( ability ) {
+
+    let difficulty = 0;
+    let currentTarget = game.user.targets.first()?.actor;
+    let currentTargets = [...game.user.targets.map( ( t ) => t.actor )];
+    let numTargets = game.user.targets.size;
+    let targetDifficultySetting = ability.system.difficulty.target
+    let groupDiffciultySetting = ability.system.difficulty.group
+    let fixedDifficultySetting = ability.system.difficulty.fixed
+
+    if ( numTargets <= 0 || targetDifficultySetting === "none" ) {
+      if ( fixedDifficultySetting > 0 ) {
+        difficulty = fixedDifficultySetting;
+      } else {
+        difficulty = 0; 
+      }
+      
+    } else {
+      let baseDifficulty = 0;
+      let additionalTargetDifficulty = 0;
+
+      if ( fixedDifficultySetting > 0 ) {
+        difficulty = fixedDifficultySetting;
+      }
+      else if ( groupDiffciultySetting !== "none" ) {
+        switch ( groupDiffciultySetting ) {
+          case 'hightestX':
+            additionalTargetDifficulty = numTargets - 1;
+          case 'highestOfGroup':
+            baseDifficulty = _getAggregatedDefense(currentTargets, targetDifficultySetting, Math.max);
+            break;
+          case 'lowestX':
+            additionalTargetDifficulty = numTargets - 1;
+          case 'lowestOfGroup':
+            baseDifficulty = _getAggregatedDefense(currentTargets, targetDifficultySetting, Math.min);
+            break;
+        }
+        difficulty = baseDifficulty + additionalTargetDifficulty;
+      } else {
+        difficulty = currentTarget?.system.characteristics.defenses[targetDifficultySetting].value ?? 0;
+      }
+
+        
+      }
+
+      /**
+       * @param { Array } targets array of all targets
+       * @param { string } targetDefenseType defense 
+       * @param { any } aggregate ???
+       * @returns { number} return  
+       */
+      function _getAggregatedDefense( targets, targetDefenseType, aggregate = Math.max ) {
```

Comment:
Doesn't belong in the Actor Document. This is some static method for combat or interaction workflows. We need to think where it makes sense.

Generally as a rule of thumb: If you don't need to use `this` in the function, it probably doesn't need to be a class method. (Except static methods which have some other connection to the class)